src/lib.rs

```diff
@@ -369,6 +369,11 @@ impl<Types: IpfsTypes> Ipfs<Types> {
     }
 
     /// Puts a block into the ipfs repo.
+    ///
+    /// # Forget safety
+    ///
+    /// Forgetting the returned future will not result in memory unsafety, but it can
+    /// deadlock other tasks.
```

Comment:
Is there any way to force a deadlock here in a test so that we can reason about / deal with it?

Reply:
I don't think we can fix this ever. This probably holds true for most synchronizing async rust code, but since I added the path which will definitely suffer from forget's I felt like I should add this here.